tests/test-cases/gitlab-runner-emulation/integration.emulate-runner.test.ts

```diff
@@ -0,0 +1,113 @@
+import {WriteStreamsMock} from "../../../src/write-streams";
+import {handler} from "../../../src/handler";
+import {initSpawnSpy} from "../../mocks/utils.mock";
+import {WhenStatics} from "../../mocks/when-statics";
+import {Utils} from "../../../src/utils";
+import {
+    GitlabRunnerCPUsPresetValue,
+    GitlabRunnerMemoryPresetValue,
+    GitlabRunnerPresetValues,
+} from "../../../src/gitlab-preset";
+
+beforeAll(() => {
+    initSpawnSpy([...WhenStatics.all, {
+        cmdArgs: ["docker", "cp", expect.any(String), expect.any(String)],
+        returnValue: {stdout: "Ok"},
+    }]);
+});
+
+test("--emulate some_unexisting_runner", async () => {
+    try {
+        const writeStreams = new WriteStreamsMock();
+        await handler({
+            cwd: "tests/test-cases/container-executable",
+            emulate: "some_unexisting_runner",
+        }, writeStreams);
+
+        expect(true).toBe(false);
+    } catch (e: unknown) {
+        expect((e as Error).message).toContain("Invalid gitlab runner to emulate.");
+    }
+});
+
+test("should contains memory config when emulating valid runner", async () => {
+    // given a spy of the bash utils
+    const bashSpy = jest.spyOn(Utils, "bash").mockResolvedValue({stdout: "abcde12345", stderr: "", exitCode: 0});
```

Comment:
Lets use the helper for mocking
https://github.com/firecow/gitlab-ci-local/blob/70f4b74ee74edeb3c6806a926169ed41b334f6fd/tests/test-cases/remote-variables-file/integration.project-variables-file.test.ts#L13